### PR TITLE
Enhance operator tests.

### LINKED ISSUE #863: Enhance operator tests.
The operator integration tests were added in https://github.com/kubearchive/kubearchive/issues/780. This was a start but not complete.

The following should be done to enhance/expand these tests.

- New tests should be added that add a global `KubeArchiveConfig`  and a local one.
- The tests should check to make sure the "sink-filters" `ConfigMap` contains the correct number of entries.
- Move the cleanup method of the integration tests to a shared method in `integration/main.go`.

### PR DESCRIPTION
**Pull request by ggallen**
## Which Issue(s) this Pull Request Resolves

Resolves #863

## Release Note

```release-note
NONE
```

## Notes for Reviewers

Also refactored the integration tests and added some speed enhancements.

The current test run had the integration tests taking 78 seconds to run.

### CODE REVIEW

File: test/integration/operator_test.go [hunk 1]
```diff
@@ -29,14 +32,13 @@ const (
...
-	a13eName    = kacName + "-a13e"
-	sinkName    = kacName + "-sink"
+	a13eName    = test.KACName + "-a13e"
+	sinkName    = test.KACName + "-sink"
 	filtersName = "sink-filters"
 )
 
 func TestEmptyKAC(t *testing.T) {
-	runTest(t, emptyKAC, map[string]any{})
+	runTest(t, "emptyKAC", map[string]any{})
```
Comment (maruiz93) [inline test/integration/operator_test.go:41]: Is there any reason to use this approach for running different versions of a test instead of the one that we use in the unit tests with [a map to store the different test cases](https://go.dev/wiki/TableDrivenTests#using-a-map-to-store-test-cases)?

I think it's the standard way to put different use cases of the same "feature" and it should be a small change from what you have right now.

File: test/integration/operator_test.go [hunk 2]
```diff
@@ -210,18 +166,66 @@ func checkResourcesAfterDelete(t testing.TB, namespace string, testName Test) {
...
+			{
+				"selector": map[string]string{
+					"apiVersion": "v1",
+					"kind":       "Pod",
+				},
+				"archiveWhen":  "status.phase == 'Succeeded'",
+			},
+		},
+	}
+	localres := map[string]any{
```
Comment (rh-hemartin) [inline test/integration/operator_test.go:186]: My local tests indicate that if the localres is empty, the local namespace is not configured for archival because it is not included in the `sink-filters` so the sink does not know that it should archive anything there, as if was not enabled in the first place. So add a test to confirm this, so with an empty `localres`.

Comment (ggallen) [reply test/integration/operator_test.go:186]: Good catch. I think you're right. I'll check this as soon as I get back.

Comment (ggallen) [reply test/integration/operator_test.go:186]: This will be handled in https://github.com/kubearchive/kubearchive/issues/948.

File: test/integration/operator_test.go (hunk 1 above)
Comment (ggallen) [reply test/integration/operator_test.go]: @maruiz93, the change is made. Can you review again?

**Review by rh-hemartin — APPROVED** [review]

**Review by maruiz93 — APPROVED** [review]

File: test/integration/operator_test.go (hunk 1 above)
Comment (maruiz93) [reply test/integration/operator_test.go]: lgtm!
